File: generate_image.js
const puppeteer = require('puppeteer');

(async () => {
  const browser = await puppeteer.launch({ headless: true });
  const page = await browser.newPage();

  // HTML content with a gradient and text overlay
  const htmlContent = `
    <html>
      <head>
        <style>
          body {
            margin: 0;
            height: 100vh;
            display: flex;
            justify-content: center;
            align-items: center;
            background: linear-gradient(to bottom right, #FF6347, #4682B4);
            font-family: Arial, sans-serif;
            color: white;
          }
          .text-overlay {
            text-align: center;
            font-size: 36px;
            font-weight: bold;
            padding: 20px;
            background: rgba(0, 0, 0, 0.5);
            border-radius: 10px;
            box-shadow: 0 4px 6px rgba(0, 0, 0, 0.2);
          }
        </style>
      </head>
      <body>
        <div class="text-overlay">
          Samsung phones can now flip into two because they are the best
        </div>
      </body>
    </html>
  `;

  // Set content to the page
  await page.setContent(htmlContent);

  // Take a screenshot of the rendered content
  await page.screenshot({ path: 'output_image.png' });

  await browser.close();
})();

File: generate_image.py
from manim import *

class GradientTextScene(Scene):
    def construct(self):
        # Set up the background gradient
        gradient = Rectangle(width=10, height=6)
        gradient.set_fill(WHITE, opacity=1)  # Fill with white
        gradient.set_color_by_gradient(RED, BLUE)  # Apply a gradient from red to blue
        self.add(gradient)

        # Create the text with beautiful font and styling
        text = Text("Samsung phones can now flip into two\nbecause they are the best", font="BeCine", font_size=36)
        text.set_color(WHITE)  # Set text color to white
        text.move_to(ORIGIN)  # Center the text in the scene
        
        # Add the text to the scene
        self.add(text)

        # Render the image
        self.wait(2)
